src/routes.js

```diff
@@ -122,6 +124,13 @@ module.exports = [
     }
   },
   {
+    method: 'POST',
+    path: '/finished',
+    handler: (request, reply) => {
+      reply(request.payload.test ? '' : request.payload);
```

Comment:
Maybe that was happening at one point, but what you have now does not look like `test` will effect if a mail is sent.. ?

You could use `NODE_ENV` within you send mail function / module, to stop it sending a mail in non production.